activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -191,19 +216,11 @@ def explain(arel, binds = [])
         MySQL::ExplainPrettyPrinter.new.pp(result, elapsed)
       end
 
-      def clear_cache!
-        super
-        reload_type_map
-      end
-
       # Executes the SQL statement in the context of this connection.
       def execute(sql, name = nil)
         log(sql, name) { @connection.query(sql) }
       end
 
-      # Mysql2Adapter doesn't have to free a result after using it, but we use this method
-      # to write stuff in an abstract way without concerning ourselves about whether it
-      # needs to be explicitly freed or not.
       def execute_and_free(sql, name = nil) # :nodoc:
```

Comment:
We can remove this method here in a different PR. This abstract class can be removed too since we only have one concrete class.